pkg/flow/internal/controller/component.go

```diff
@@ -63,18 +65,18 @@ type DialFunc func(ctx context.Context, network, address string) (net.Conn, erro
 // ComponentGlobals are used by ComponentNodes to build managed components. All
 // ComponentNodes should use the same ComponentGlobals.
 type ComponentGlobals struct {
-	LogSink           *logging.Sink                // Sink used for Logging.
-	Logger            *logging.Logger              // Logger shared between all managed components.
-	TraceProvider     trace.TracerProvider         // Tracer shared between all managed components.
-	Clusterer         *cluster.Clusterer           // Clusterer shared between all managed components.
-	DataPath          string                       // Shared directory where component data may be stored
-	OnComponentUpdate func(cn *ComponentNode)      // Informs controller that we need to reevaluate
-	OnExportsChange   func(exports map[string]any) // Invoked when the managed component updated its exports
-	Registerer        prometheus.Registerer        // Registerer for serving agent and component metrics
-	HTTPPathPrefix    string                       // HTTP prefix for components.
-	HTTPListenAddr    string                       // Base address for server
-	DialFunc          DialFunc                     // Function to connect to HTTPListenAddr.
-	ControllerID      string                       // ID of controller.
+	Logger               *logging.Logger                            // Logger shared between all managed components.
+	TraceProvider        trace.TracerProvider                       // Tracer shared between all managed components.
+	Clusterer            *cluster.Clusterer                         // Clusterer shared between all managed components.
+	DataPath             string                                     // Shared directory where component data may be stored
+	OnComponentUpdate    func(cn *ComponentNode)                    // Informs controller that we need to reevaluate
+	OnExportsChange      func(exports map[string]any)               // Invoked when the managed component updated its exports
+	Registerer           prometheus.Registerer                      // Registerer for serving agent and component metrics
+	HTTPPathPrefix       string                                     // HTTP prefix for components.
+	HTTPListenAddr       string                                     // Base address for server
+	DialFunc             DialFunc                                   // Function to connect to HTTPListenAddr.
+	ControllerID         string                                     // ID of controller.
+	ModuleControllerFunc func(id string) component.ModuleController // Func to generate a module controller.
```

Comment:
nit: I was confused what this function did at first. `NewModuleController` might help make it more clear.